packages/core/src/utils.ts

```diff
@@ -243,3 +250,59 @@ export async function getRawConfiguration(
   }
   return sygmaConfig;
 }
+
+/**
+ * Validate Substrate address.
+ * @param {string} address
+ */
+export function isValidSubstrateAddress(address: string): boolean {
+  try {
+    encodeAddress(isHex(address) ? hexToU8a(address) : decodeAddress(address));
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * Validate EVM address.
+ * @param {string} address
+ * @returns {boolean}
+ */
+export function isValidEvmAddress(address: string): boolean {
+  return !!ethers.utils.isAddress(address);
+}
+
+/**
+ * Validate Bitcoin address.
+ * @param {string} address
+ * @returns {boolean}
+ */
+export function isValidBitcoinAddress(address: string): boolean {
+  if (process.env.SYGMA_ENV === Environment.TESTNET || process.env.SYGMA_ENV === Environment.DEVNET)
+    return validate(address, BitcoinNetwork.testnet);
+
+  return validate(address, BitcoinNetwork.mainnet);
```

Comment:
I think there is a way to validate the address without the need of a new dependency. We can still leverage btc.js dependency for this. 

you can run this method from btc.js lib:

```ts
try {
     const outputScript = bitcoin.address.toOutputScript(address, network);
} catch (e) {
   // ..some error handling here
}
```

`network` should be the network object from the btc dependency which you are already using

If you want to be sure, you can use the inverse method:

```ts
bitcoin.address.fromOutputScript(Buffer.from(yourOutputScript, 'hex'), network)
```

This will give you back the address that you used above for validation purposes

Reply:
@wainola The code is working for all types of addresses except P2TR, I found some explanation why and tried the alternative solution provided in the link, but it didn't work as well. Maybe you know something specific why it might happen? https://github.com/bitcoinjs/bitcoinjs-lib/issues/1919

Reply:
ok after some discussion about this with @Lykhoyda we realized that there is a problem of using `bitcoin` lib to validate address which is that for this to work you need to init btc.js with tinysecp from `tiny-secp256k1`, which is not ideal to have inside the SDK. We can go with the original proposed solution in the meantime which avoid the introduction of a third dep -` tinysecp`- just for the sake of validating the address here